ui/receive_page.go

```diff
@@ -0,0 +1,594 @@
+			return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+				layout.Rigid(func(gtx C) D {
+					return layout.Inset{Bottom: values.MarginPadding16}.Layout(gtx, func(gtx C) D {
+						return pg.topNav(gtx, common)
+					})
+				}),
+				layout.Rigid(func(gtx C) D {
+					return common.theme.Card().Layout(gtx, func(gtx C) D {
+						return pg.pageContainer.Layout(gtx, len(pageContent), func(gtx C, i int) D {
+							return pageContent[i](gtx)
+						})
+					})
+				}),
+			)
+		})
+	})
+
+	if pg.wallAcctWidget.isWalletAccountModalOpen {
+		return common.Modal(gtx, dims, pg.wallAcctWidget.walletAccountSection(gtx, common))
+	}
+
+	return dims
+}
+
+func (pg *receivePage) pageSections(gtx layout.Context, body layout.Widget) layout.Dimensions {
+	return pg.theme.Card().Layout(gtx, func(gtx C) D {
+		gtx.Constraints.Min.X = gtx.Constraints.Max.X
+		return layout.UniformInset(values.MarginPadding16).Layout(gtx, body)
+	})
+}
+
+func (pg *receivePage) pageBackDrop(gtx layout.Context) {
+	if pg.isNewAddr {
+		gtx.Constraints.Min.X = gtx.Constraints.Max.X
+		gtx.Constraints.Min.Y = gtx.Constraints.Max.Y
+		m := op.Record(gtx.Ops)
+		pg.backdrop.Layout(gtx)
+		op.Defer(gtx.Ops, m.Stop())
+	}
+}
+
+func (pg *receivePage) topNav(gtx layout.Context, common pageCommon) layout.Dimensions {
+	m := values.MarginPadding20
+	return layout.Flex{}.Layout(gtx,
+		layout.Rigid(func(gtx C) D {
+			return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+				layout.Rigid(func(gtx C) D {
+					common.subPageBackButton.Icon = common.icons.contentClear
+					return common.subPageBackButton.Layout(gtx)
+				}),
+				layout.Rigid(func(gtx C) D {
+					txt := common.theme.H6("Receive DCR")
+					txt.Color = common.theme.Color.DeepBlue
+					return layout.Inset{Left: m}.Layout(gtx, txt.Layout)
+				}),
+			)
+		}),
+		layout.Flexed(1, func(gtx C) D {
+			return layout.E.Layout(gtx, func(gtx C) D {
+				return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+					layout.Rigid(func(gtx C) D {
+						return common.subPageInfoButton.Layout(gtx)
+					}),
+				)
+			})
+		}),
+	)
+}
+
+func (pg *receivePage) generateAddressSection(gtx layout.Context, common pageCommon) layout.Dimensions {
+	return layout.Flex{Spacing: layout.SpaceBetween}.Layout(gtx,
+		layout.Rigid(func(gtx C) D {
+			return common.theme.Body1("Your Address").Layout(gtx)
+		}),
+		layout.Rigid(func(gtx C) D {
+			return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+				layout.Rigid(func(gtx C) D {
+					if pg.isNewAddr {
+						m := op.Record(gtx.Ops)
+						layout.Inset{Top: values.MarginPadding30, Left: unit.Dp(-152)}.Layout(gtx, func(gtx C) D {
+							return pg.newAddr.Layout(gtx)
+						})
+						op.Defer(gtx.Ops, m.Stop())
+					}
+					return layout.Dimensions{}
+				}),
+				layout.Rigid(func(gtx C) D {
+					return pg.more.Layout(gtx)
+				}),
+			)
+		}),
+	)
+}
+
+func (pg *receivePage) addressSection(gtx layout.Context, common pageCommon) layout.Dimensions {
```

Comment:
This is a row not a section. A section could be demarcated with a separator or space between different card layouts. Other methods that this applies to should be addressed as well.